### PR TITLE
Solr 8 upgrade changes.

### PR DESCRIPTION
**Pull request by dkinzer**
* Replaces removed analyzers.
* Updates Lucene version match to 7.0.0
* Adds two new searchers title_search, subject_search because these are needed
for title and subject searches.

### CODE REVIEW

File: solrconfig.xml [hunk 1]
```diff
@@ -237,6 +237,208 @@
...
+         url_finding_aid_display:[json],
+         url_more_links_display:[json],
+         electronic_resource_display:[json]
+       </str>
+
+       <str name="facet">true</str>
+       <str name="facet.mincount">1</str>
+       <str name="facet.field">format</str>
+       <str name="facet.field">subject_facet</str>
+       <str name="facet.field">availability_facet</str>
```
Comment (sensei100) [inline solrconfig.xml]: @dkinzer Is there any reason for having only these particular facets defined here?  I'm just wondering why these and not the rest of the facets used.

File: solrconfig.xml [hunk 2]
```diff
@@ -237,6 +237,208 @@
...
+         subject_facet^1000000.0
+       </str>
+       <int name="ps">3</int>
+       <float name="tie">0.01</float>
+
+       <!-- NOT using marc_display because it is large and will slow things down for search results -->
+       <str name="fl">
+         id,
+         format,
+         database_display,
```
Comment (sensei100) [inline solrconfig.xml]: Do we have a database_display field in the catalog?  I don't see it defined in the controller.

File: solrconfig.xml (hunk 1 above)
Comment (dkinzer) [reply solrconfig.xml]: @sensei100 I copied this directly from the "search" requestHandler above for parity with that one.  I'm still looking into what technique can be used to inherit from another requestHandler with just minor overrides.

File: solrconfig.xml (hunk 2 above)
Comment (sensei100) [reply solrconfig.xml]: Sorry, I just realized this IS the AZ solr!

**Pull request merged**